source/faq.txt

```diff
@@ -261,31 +261,53 @@ socket connections.
 If you experience unexpected network behavior, here
 are some things to check:
 
-#. The firewall should send a ``FIN packet`` when closing a socket,allowing the driver to detect that the socket is closed.
+#. The firewall should send a ``FIN packet`` when closing a socket,
+   allowing the driver to detect that the socket is closed.
 #. The firewall should allow ``keepAlive`` probes.
 
 How Can I Prevent a Slow Operation From Delaying Other Operations?
 ------------------------------------------------------------------
 
-To control the maximum size of a connection pool, you can set the
-``maxPoolSize`` option in the :ref:`connection options
-<node-connection-options>`. The default value of ``maxPoolSize`` is
-``100``. If the number of in-use connections to a server reaches
-``maxPoolSize``, the next request to that server will wait 
-until a connection becomes available. To prevent long-running operations
-from slowing down your application, you can increase ``maxPoolSize``.
-
-The driver does not limit the number of requests that can wait for
-sockets to become available. Requests wait for the amount of time
-specified in the ``waitQueueTimeoutMS`` option, which
-defaults to ``0`` (no limit). You should set this option if it is
-more important to stop long-running operations than it is to complete
-every operation.
+Before you make changes to your connection configuration or application,
+you should determine which operations could be causing delays. You can
+investigate the performance of in-progress operations by using the
+following strategies:
+
+- Enable the database profiler on your deployment. To learn more, see
+  :manual:`Database Profiler </tutorial/manage-the-database-profiler/>`
+  in the Server manual.
+- Run the ``db.currentOp()`` :manual:`shell command </reference/method/>`. To learn more, see the 
+  :manual:`db.currentOp() </reference/method/db.currentOp/>`
+  documentation in the Server manual.
+- Enable connection pool monitoring. To learn more, see
+  :ref:`node-connection-pool-monitoring`.
+
+Once you determine how and why your operations are delayed, if possible,
+try to improve the performance of your application. Read the :website:`Best
+Practices Guide for MongoDB Performance </basics/best-practices>` for
+possible solutions.
+
+If you cannot avoid time-consuming or slow operations, you can
+modify your connection settings to increase the size of the connection
+pool, a group of connections to the server that the driver maintains at
+any time.
+
+To specify the maximum size of a
+connection pool, you can set the ``maxPoolSize`` option in the
+:ref:`connection options <node-connection-options>`. The default value
+of ``maxPoolSize`` is ``100``. If the number of in-use connections to a
+server reaches ``maxPoolSize``, the next operation sent to the server
+will wait until a connection becomes available. The following code sets
```

Comment:
Issue:

Regarding https://github.com/mongodb/docs-node/pull/728#discussion_r1275260135

I think the "operation waits" is not an accurate characterization of an operation. I think an operation should either be the driver sending the command that corresponds to the operation or the server processing/returning a response to an operation.

Suggestion:
If it happens to be the driver waiting to send it to the server, maybe this would describe it:
"the next operation the driver attempts to run on the server needs to wait for an available connection."

If it is the server waiting on the driver because the command has already been sent and executed:
"the operation sent to the server pauses until a free connection to the driver becomes available."

Reply:
From the way it is described in the spec and the manual documentation, it seems like this is a Server-side limit. That is, the driver sends the request/executes the operation, but it waits on the server side until a connection is freed